### DISCUSSION TITLE
Custom Views (for ListOperation) - View not showing

### DISCUSSION THREAD

**Discussion by Read2uk** [Laravel-Backpack/community-forum#1173] [Q&A (Help)] [answered]
Hi all,

Just a quick question, not sure if I am doing something wrong. Basically want to add Custom Views but the dropdown button is not showing the added views (is not clickable on the dropdown)

This is what the CrudController looks like:

![image](https://github.com/user-attachments/assets/830a0699-078a-4fa5-8c1c-7abf37b940fa)

![image](https://github.com/user-attachments/assets/d912567e-502c-46bb-b6d4-694d1d705522)

![image](https://github.com/user-attachments/assets/347ad5c6-8dc3-4c23-b932-f4900f616aa9)

The 'Views' dropdown is not working, so can't select the other views

![image](https://github.com/user-attachments/assets/0c8a06ad-861d-4f14-a517-1259b2550efd)

**Comment by Read2uk** [comment]
On the last screenshot there is a log warning:

LOG.warning: method_exists(): Passing null to parameter #2 ($method) of type string is deprecated in C:\localhost\test\vendor\backpack\pro\src\Http\Controllers\Operations\CustomViewOperation.php on line 101

Not sure if its related to this?

**Comment by jcastroa87** [comment]
Hello @Read2uk 

Can you please share more information about this.

Versions
```
php artisan backpack:version
```

Then the way how you need to set View Method is with a query, like this.
```
public function setupLast5YearsView()
    {
        CRUD::addClause('where', 'issuance_date', '>=', now()->subYears(5)->format('Y-m-d'));
    }
```

If still doesn't work, try to run just `$this->runCustomViews();`so Backpack will get the Custom Views in Controller.

Cheers.

**Reply by Read2uk** [reply]
Hi @jcastroa87 

backpack/crud: 6.7.38
backpack/generators: v4.0.6
backpack/logmanager: v5.0.2
backpack/permissionmanager: 7.2.1
backpack/pro: 2.2.17
backpack/theme-coreuiv2: 1.2.5

Also:
Backpack 6 (migrated from 5)
PHP 8.3 (migrated from 8.1)
Laravel 11 (migrated from 10)


So I tried what you suggested, still not working:

![image](https://github.com/user-attachments/assets/e1730bc0-ae15-486d-9bbf-eb05e73df7ff)

The Views dropdown is not working.
![image](https://github.com/user-attachments/assets/5adecc5d-9f7a-47cb-a983-82b6719a99b1)
I do get a warning when I load the view:
LOG.warning: method_exists(): Passing null to parameter https://github.com/Laravel-Backpack/community-forum/issues/2 ($method) of type string is deprecated in C:\localhost\test\vendor\backpack\pro\src\Http\Controllers\Operations\CustomViewOperation.php on line 101


Weirdly enough, I do see the URL in the source code for the custom view http://localhost:8000/assignment/view/last5-years 

![image](https://github.com/user-attachments/assets/183b87d3-11c1-4e05-a29e-be156c99ba25)

**Reply by jcastroa87** [reply]
Did you try this in a clean installation? Because look like a problem in the upgrade, i try in my side and there is no problem.

Cheers.

**Reply by Read2uk** [reply]
Do you mean in a clean installation of Laravel and Backpack? No because I already have an app running on backpack 5.

Do you have the same packages when you tired at your end?

backpack/crud: 6.7.38
backpack/generators: v4.0.6
backpack/logmanager: v5.0.2
backpack/permissionmanager: 7.2.1
backpack/pro: 2.2.17
backpack/theme-coreuiv2: 1.2.5

Also:
Backpack 6 (migrated from 5)
PHP 8.3 (migrated from 8.1)
Laravel 11 (migrated from 10)

**Reply by pxpm** [reply]
Hey @Read2uk I got the error you reported. 

I've just released 2.2.18 of `backpack/pro` with the fix. 

Please allow sometime for our private repository to pick the changes and you can get it with a `composer update`. 

Please let us know if you are still experience this issue after the update 👍

**Reply by Read2uk** [reply]
Thanks @pxpm  can I ask if its only affecting when users upgrades from v5 to v6?

Whats the ETA for 2.2.18?

Thanks

**Reply by pxpm** [reply]
The job that syncs the changes run every couple hours, so it does not take long. it's already available: https://repo.backpackforlaravel.com/#backpack/pro

The users affected are PHP 8.3 users, not related specific to backpack version 👍 

Let me know if it's working on your side. 

Cheers

**Reply by Read2uk** [reply]
Hi,

Still not working on my side, although this warning its gone away:

 LOG.warning: method_exists(): Passing null to parameter https://github.com/Laravel-Backpack/community-forum/issues/2 ($method) of type string is deprecated in C:\localhost\test\vendor\backpack\pro\src\Http\Controllers\Operations\CustomViewOperation.php on line 101


This is still showing in debugbar: LOG.warning: ctype_lower(): Argument of type bool will be interpreted as string in the future in C:\localhost\test\vendor\laravel\framework\src\Illuminate\Support\Str.php on line 1481

**Reply by Read2uk** [reply]
Here is another screenshot of the Custom Views appearing on the source code:

![image](https://github.com/user-attachments/assets/3d8a9361-a065-4896-98b1-6b402130d117)

**Reply by Read2uk** [reply]
Hi @pxpm @jcastroa87 

So I have actually found the problem but it needs fixing at your end.

I have checked your demo, and its also broken.

https://demo.backpackforlaravel.com/admin/pet-shop/invoice
![image](https://github.com/user-attachments/assets/4bc62628-fd4a-49c6-9e00-4871a203878b)


![image](https://github.com/user-attachments/assets/6b9ff165-96d3-4141-90b8-ff76cf66c157)


I am using Core UI v2 Theme which is Bootstrap 4  but the dropdown for the view is using **data-bs-toggle** which is Bootstrap 5, as soon as I make it into bootstrap 4  **data-toggle** it works.

**Reply by pxpm** [reply]
Hey @Read2uk 

We are aware of the "differences" between BS4 and BS5, are you sure you are not overwriting the `datatables_logic` into your application resources folder ? 

Because we use both `data-bs-toggle` and `data-toggle` and it works with both versions of BS (bootstrap) as you can see here: https://github.com/Laravel-Backpack/CRUD/blob/b58f971aa0f0064f5070ae51f90249d9675c2d50/src/resources/views/crud/inc/datatables_logic.blade.php#L433

Let me know if that was the issue, if not I will try to debug it further, but I am pretty sure it should work, as @jcastroa87 showed, he tested it using coreuiv2 too. 

Cheers

Edit: further investigation, it seems it does not pick both. I will investigate further tomorrow. Thanks for the heads up.

**Reply by pxpm** [reply, marked as answer]
I've just released backpack/pro 2.2.19 with the fix. 

In a couple of hours our private repository should pick the change and you can get the fixed version with a `composer update`. 

Once again, thank you very much for the report, and for the debug process, we really appreciate it 🙏 

Cheers

**Reply by Read2uk** [reply]
Thanks for your help, I updated the packages and I can confirm its now fixed!
